packages/astro/src/vite-plugin-astro/compile.ts

```diff
@@ -34,6 +34,17 @@ export async function compileAstro({
 
 	try {
 		transformResult = await compile(compileProps);
+
+		// Check for client: hydration directives in the compiled JSX structure
+		const hasHydrationDirective = transformResult.code.includes('client:');
+		if (hasHydrationDirective) {			
+			// Optionally log a warning or error for devs
+			logger.warn(
+				null,
+				`Hydration directive found in ${compileProps.filename}. Astro components should not use client-side rendering.`
```

Comment:
Rn we show in what component an invalid client directive is found right? Is it also possible to show on what component it happens?

Reply:
from my understanding the error is from the way the component is used. and the file that calls it incurrectly that is logged.